src/tuxpaint-config.1
.\" tuxpaint-config.1 - 2008.07.14
.TH TUXPAINT-CONFIG 1 "14 July 2008" "0.0.12" "Tux Paint Configuration Tool"
.SH NAME
tuxpaint-config -- A configuration tool for Tux Paint, a drawing program for young children.

.SH SYNOPSYS
.B tuxpaint-config
[\-\-help \-\-version \-\-usage \-\-copying]

.SH DESCRIPTION
.PP
\fITux Paint Configuration Tool\fP (tuxpaint-config) is a graphical
configuration program for Tux Paint, a drawing program for young children.
It can be used to create a configuration file for Tux Paint, rather than
doing it manually by editing the configuration file in a text edtior.

.SH OPTIONS - INFORMATIONAL
.TP 8
.B \-\-help
Display short, helpful information about Tux Paint.
.TP 8
.B \-\-version
Output the version info.
.TP 8
.B \-\-usage
Display a list of all commandline options.
.TP 8
.B \-\-copying
Show the license (GNU GPL) under which Tux Paint is released.

.SH OPTIONS - LANGUAGE
.l
Much like Tux Paint itself, \fITux Paint Configuration Tool\fP has been
translated into numerous languages.  \fItuxpaint-config\fP will try its
best to honor your \fIlocale\fP setting (i.e., the \fILANG\fP environment
variable), if possible.

.SH ENVIRONMENT
.l
.PP
.TP 8
.B HOME
to determine where the user's configuration file is.

.TP 8
.B LANG
to determine langauge to use, if \fIsetlocale(3)\fP refers to 'LC_MESSAGES'.

.SH FILES
.TP 8
.B /etc/tuxpaint/tuxpaint.conf
System-wide configuration file.
.RS
.PP
(Created during installation.)
.RE
.TP 8
.B $HOME/.tuxpaintrc
User's configuration file.  It can be used to set default options
(rather than setting them on the command-line every time), and/or to
override any settings in the system-wide configuration file.
.RS
.PP
.RE

.SH COPYRIGHT
This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 2 of the License, or
(at your option) any later version.

.SH OTHER INFO
The canonical place to find \fITux Paint\fP information is at
http://www.tuxpaint.org/.

.SH AUTHORS
Torsten Giebl,
Jan Wynholds,
Bill Kendrick,
Martin Fuhrer.

.SH "SEE ALSO"
.BR tuxpaint (1)
.PP
And documentation within /usr/[local/]share/doc/tuxpaint-config/.



debian/changelog
tuxpaint-config (0.0.13-8+lliurex3) bionic; urgency=medium

  * Patched for missing string.h header, strlen function not recognized

 -- Juan Ramon Pelegrina <juapesai@hotmail.com>  Tue, 16 Oct 2018 09:40:52 +0200

tuxpaint-config (0.0.13-8) unstable; urgency=medium

  * Move git repository to salsa.debian.org
  * Update standards version to 4.1.3
  * Update compat level to 11
  * Add Debian Junior team to uploaders

 -- Jonathan Carter <jcc@debian.org>  Wed, 14 Feb 2018 11:53:42 +0200

tuxpaint-config (0.0.13-7) unstable; urgency=medium

  * Fix parallel builds (Closes: #879042)

 -- Jonathan Carter <jcc@debian.org>  Mon, 13 Nov 2017 15:07:32 +0200

tuxpaint-config (0.0.13-6) unstable; urgency=medium

  * Update standards version to 4.1.1
    - Update copyright file to copyright format 1.0
    - Remove obsolete menu file
    - Remove obsolete -k from dh_clean
    - Set hardening=+all
    - Remove ancient conflicts
  * Fix spelling mistakes in man page
  * debian/rules:
    - Remove boilerplate text
  * Add -fpic for x32

 -- Jonathan Carter <jcc@debian.org>  Mon, 30 Oct 2017 09:20:13 +0200

tuxpaint-config (0.0.13-5) unstable; urgency=medium

  * Fix cross-builds (Closes: #878365)

 -- Jonathan Carter <jcc@debian.org>  Sat, 28 Oct 2017 22:11:39 +0200

tuxpaint-config (0.0.13-4) unstable; urgency=medium

  * Adopting package (Closes: #848989)
  * Remove boilerplate from include-standard-flags patch
  * Update compat to 10
  * Update copyright file to copyright-format 1.0
  * debian/rules:
     - Remove boilerplate comments from debian/rules
     - Set DEB_BUILD_MAINT_OPTIONS=hardening=+all
     - Use build architecture toolchain (Closes: #878365)
  * debian/control:
     - Remove ancient Conflicts
     - Update standards version to 4.1.1

 -- Jonathan Carter <jcc@debian.org>  Wed, 18 Oct 2017 15:08:25 +0200

tuxpaint-config (0.0.13-3) unstable; urgency=medium

  * QA upload.
  * Imported source into a gbp repo in collab-maint.
  * Removed empty /usr/share/applnk/ from binary package.
  * Changed Standards-Version from 3.9.6 to 4.1.0.
  * Added homepage link to d/control.
  * Removed obsolete menu file.
  * Changed to debhelper level 9.
  * Added gbp.conf file to enfore use of pristine-tar.
  * Added watch file pointing to sourceforge.

 -- Petter Reinholdtsen <pere@debian.org>  Wed, 30 Aug 2017 13:31:19 +0200

tuxpaint-config (0.0.13-2) unstable; urgency=medium

  * QA upload.
  * Set maintainer to Debian QA Group. (see #848989)
  * Add patch from Chris Lamb to make the build reproducible.
    (Closes:  #834780)

 -- Adrian Bunk <bunk@debian.org>  Sun, 22 Jan 2017 01:48:06 +0200

tuxpaint-config (0.0.13-1) unstable; urgency=low

  * New upstream release. (Closes: #626782, #722947)
  * Switch to dpkg-source 3.0 (quilt) format.
  * Include watch file, thanks to Daniel Liedert <daniel.leidert@wgdd.de>.
    (Closes: #549385).
  * Update watch file to new location.
  * Update to current standards.

 -- Ben Armstrong <synrg@sanctuary.nslug.ns.ca>  Sun, 26 Oct 2014 09:53:13 -0300

tuxpaint-config (0.0.12-3) unstable; urgency=low

  * Eliminate non-literal format string. (Closes: #652175)

 -- Ben Armstrong <synrg@sanctuary.nslug.ns.ca>  Mon, 19 Dec 2011 05:50:56 -0400

tuxpaint-config (0.0.12-2) unstable; urgency=low

  * Remove redundant desktop file.  (Closes: #498927)

 -- Ben Armstrong <synrg@sanctuary.nslug.ns.ca>  Fri, 16 Oct 2009 09:30:06 -0300

tuxpaint-config (0.0.12-1) unstable; urgency=low

  * New upstream release.

 -- Ben Armstrong <synrg@sanctuary.nslug.ns.ca>  Sun, 09 Aug 2009 12:20:30 -0300

tuxpaint-config (0.0.10-1) unstable; urgency=low

  * New upstream release.
  * Install desktop file to opendesktop location.

 -- Ben Armstrong <synrg@sanctuary.nslug.ns.ca>  Sun, 08 Jun 2008 14:41:23 -0300

tuxpaint-config (0.0.8-1) unstable; urgency=low

  * New upstream release.

 -- Ben Armstrong <synrg@sanctuary.nslug.ns.ca>  Tue, 03 Jul 2007 22:12:27 -0300

tuxpaint-config (0.0.7+0.0.8rc1-1) unstable; urgency=low

  * New upstream release.

 -- Ben Armstrong <synrg@sanctuary.nslug.ns.ca>  Sat, 30 Jun 2007 17:46:26 -0300

tuxpaint-config (0.0.7-1) unstable; urgency=low

  * New upstream release.

 -- Ben Armstrong <synrg@sanctuary.nslug.ns.ca>  Sun, 22 Oct 2006 12:00:14 -0300

tuxpaint-config (0.0.6+0.0.7rc5-1) unstable; urgency=low

  * New upstream release.

 -- Ben Armstrong <synrg@sanctuary.nslug.ns.ca>  Sun, 15 Oct 2006 11:05:04 -0300

tuxpaint-config (0.0.6-2) unstable; urgency=low

  * Fixed X11 dependencies (thanks, Matt Kraai) (Closes: #356274)

 -- Ben Armstrong <synrg@sanctuary.nslug.ns.ca>  Sun,  2 Apr 2006 10:28:04 -0300

tuxpaint-config (0.0.6-1) unstable; urgency=low

  * New upstream release.
  * Fix missing menu entry.

 -- Ben Armstrong <synrg@sanctuary.nslug.ns.ca>  Sat, 21 Jan 2006 19:06:41 -0400

tuxpaint-config (0.0.5-1) unstable; urgency=low

  * Initial Release.

 -- Ben Armstrong <synrg@sanctuary.nslug.ns.ca>  Thu,  7 Oct 2004 10:26:42 -0300



src/tuxpaint-config.cxx
/*
  tuxpaint-config.c

  Tux Paint Configuration Tool - A GUI configuration editor for Tux Paint,
                                 a simpel drawing program for children.

  Copyright (c) 2002-2008
  
  Torsten Giebl <wizard@turricane.org>
  Jan Wynholds
  Bill Kendrick <bill@newbreedsoftware.com>

  http://www.tuxpaint.org/
  
  November 12, 2002 - July 14, 2008
*/


#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include <locale.h>

#include <libintl.h>
#ifndef gettext_noop
#define gettext_noop(String) String
#endif

#if !defined (__WIN32__) && !defined(__APPLE__) && !defined(__BEOS__)
#include <paper.h>
#endif

#include "version.h"
#include "about.h"


#ifdef WIN32
#define LOCALEDIR "locale"
#endif

#include <errno.h>

#include <FL/Fl.H>
#include <FL/Fl_Window.H>
#include <FL/Fl_Box.H>

#if !defined(Fl_H)
#error "---------------------------------------------------"
#error "If you installed FLTK from a package, be sure to get"
#error "the development package, as well!"
#error "(e.g., 'libfltk1-dev.rpm')"
#error "---------------------------------------------------"
#endif

#include "tuxpaint-config2.h"

/* Local function prototypes: */

void setup(int argc, char * argv[]);
int mainloop(int argc, char * argv[]);


int main(int argc, char * argv[])
{
  int res;

  setup(argc, argv);

  res = mainloop(argc, argv);

#if !defined (__WIN32__) && !defined(__APPLE__) && !defined(__BEOS__)
  paperdone();
#endif

  return res;
}


int mainloop(int argc, char * argv[])
{
    Fl::scheme ("none");	
    Show_TPC_Window ();

    return Fl::run ();
}


void setup(int argc, char * argv[])
{
  int i;
  int show_usage;
  int exit_val;
    
#if !defined (__WIN32__) && !defined(__APPLE__) && !defined(__BEOS__)
  const struct paper * ppr;
  int max_papersizes;
#endif
    
  show_usage = 0;
  exit_val = 0;

  for (i = 1; i < argc; i++)
  {
    if (strcmp(argv[i], "--help") == 0)
    {
      unsigned int x, j, len;
      char buf[80];

      buf[0] = '\0';
      len = 0;
      x = 0;

      for (j = 0; j <= strlen(about_text); j++)
      {
        buf[len++] = about_text[j];

        if (about_text[j] == ' ' || about_text[j] == '\0')
        {
          if (x + len > 78)
          {
            printf("\n");
            x = 0;
          }

          buf[len] = '\0';
          printf("%s", buf);
          x += len;

          len = 0;
        }
      }

      printf("\n\n");

      exit(0);
    }
    else if (strcmp(argv[i], "--usage") == 0)
    {
      show_usage = 1;
      exit_val = 0;
    }
    else if (strcmp(argv[i], "--copying") == 0)
    {
      /* FIXME: Copying! */
      printf("\nTux Paint Config. version " VER_VERSION ", " VER_DATE "\n\n");

      printf("This program is free software; you can redistribute it\n"
	     "and/or modify it under the terms of the GNU General Public\n"
	     "License as published by the Free Software Foundation;\n"
	     "either version 2 of the License, or (at your option) any\n"
	     "later version.\n"
	     "\n"
	     "This program is distributed in the hope that it will be\n"
	     "useful and entertaining, but WITHOUT ANY WARRANTY; without\n"
	     "even the implied warranty of MERCHANTABILITY or FITNESS\n"
	     "FOR A PARTICULAR PURPOSE.  See the GNU General Public\n"
	     "License for more details.\n"
	     "\n"
	     "You should have received a copy of the GNU General Public\n"
	     "License along with this program; if not, write to the Free\n"
	     "Software Foundation, Inc., 59 Temple Place, Suite 330,\n"
	     "Boston, MA  02111-1307  USA\n\n");
	exit(0);  
    }
    else if (strcmp(argv[i], "--version") == 0)
    {
      printf("Tux Paint Config. version " VER_VERSION ", " VER_DATE "\n");
      exit(0);
    }
    else
    {
      show_usage = 1;
      exit_val = 1;
    }
  }

  if (show_usage)
  {
    printf("Usage: %s [--version | --copying | --usage | --help]\n",
           argv[0]);

    /* Kluge to get around OS X sending arguments to apps; jdrake 2004-10-02 */
#ifndef __APPLE__
    exit(exit_val);
#endif
  }


  setlocale(LC_ALL,"");

  bindtextdomain("tuxpaint-config", LOCALEDIR);
  /* Old version of glibc does not have bind_textdomain_codeset() */
#if defined __WIN32 || __GLIBC__ && __GLIBC__ == 2 && __GLIBC_MINOR__ >=2 || __GLIBC__ > 2 || __APPLE__
  bind_textdomain_codeset("tuxpaint-config", "UTF-8");
#endif
  textdomain("tuxpaint-config");


  /* Get available paper sizes: */

#if !defined (__WIN32__) && !defined(__APPLE__) && !defined(__BEOS__)
  paperinit();

  max_papersizes = 5;
  papersizes = (char * *) malloc(sizeof (char * *) * max_papersizes);

  papersizes[0] = strdup(gettext("(Use system's default)"));
  num_papersizes = 1;
  ppr = paperfirst();
  
  do
  {
    if (ppr != NULL)
    {
      if (num_papersizes >= max_papersizes)
      {
        max_papersizes += 5;
        papersizes = (char * *) realloc(papersizes, sizeof(char * *) * max_papersizes);
      }

      papersizes[num_papersizes++] = strdup(papername(ppr));
    }

    ppr = papernext(ppr);
  }
  while (ppr != NULL);

#endif
}
